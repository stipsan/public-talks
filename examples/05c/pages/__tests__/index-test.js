import { Simulate, render } from 'react-testing-library'
import snapshotDiff from 'snapshot-diff'
import App from '../index'

it('should open and close', () => {
  const { container, getByTestId } = render(<App />)
  expect(container).toMatchSnapshot()
  const firstVersion = container.cloneNode(true)

  Simulate.click(getByTestId('toggle-modal'))
  expect(container).toMatchSnapshot()
  const secondVersion = container.cloneNode(true)

  expect(snapshotDiff(firstVersion, secondVersion)).toMatchSnapshot()
})
