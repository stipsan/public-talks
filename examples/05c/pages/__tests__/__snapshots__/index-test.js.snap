// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should open and close 1`] = `
<div>
  <div>
    <button
      data-testid="toggle-modal"
    />
  </div>
</div>
`;

exports[`should open and close 2`] = `
<div>
  <div>
    <div>
      Modal content
    </div>
    <button
      data-testid="toggle-modal"
    />
  </div>
</div>
`;

exports[`should open and close 3`] = `
"Snapshot Diff:
- First value
+ Second value

  <div>
    <div>
+     <div>
+       Modal content
+     </div>
      <button
        data-testid=\\"toggle-modal\\"
      />
    </div>
  </div>"
`;
